ddtrace/contrib/grpc/client_interceptor.py

```diff
@@ -46,37 +46,54 @@ class _ClientCallDetails(
     pass
 
 
-def _handle_response_or_error(span, response_or_error):
-    # response_of_error should be a grpc.Future and so we expect to have
+def _future_done_callback(span):
+    def func(response):
+        try:
+            # pull out status code from gRPC response to use both for `grpc.status.code`
+            # tag and the error type tag if the response is an exception
+            status_code = to_unicode(response.code())
+            span.set_tag(constants.GRPC_STATUS_CODE_KEY, status_code)
+
+            if status_code != grpc.StatusCode.OK:
+                _handle_error(span, response, status_code)
+        finally:
+            span.finish()
+
+    return func
+
+
+def _handle_response(span, response):
+    if not isinstance(response, grpc.Future):
+        return response
+
+    response.add_done_callback(_future_done_callback(span))
```

Comment:
do we need to `return response` at the end here?

Reply:
`_handle_response` does not itself get the response, though it can (as in the case of adding a callback) mutate it. the response object is gotten before this function is called and returned by the calling function.

Reply:
got it, so really we don't need to do `return response` above then, just `return` is fine?

Maybe we can change to:

```python
if isinstance(response, grpc.Future):
    response.add_done_callback(_future_done_callback(span))
```

?

Reply:
Yes, missed that above!